search/includes/classes/queue/class-cron.php

```diff
@@ -62,6 +62,13 @@ public function init() {
 		add_action( self::SWEEPER_CRON_EVENT_NAME, [ $this, 'sweep_jobs' ] );
 		add_action( self::TERM_UPDATE_CRON_EVENT_NAME, [ $this, 'queue_posts_for_term_taxonomy_id' ] );
 
+		// Make job processing = indexing of documents run concurrently. This should help with handling spikes
+		// of bulk reindexing as well as keep cron's option used for queued jobs small, by processing them faster.
+		add_filter( 'a8c_cron_control_concurrent_event_whitelist', function( $whitelist ) {
+			$whitelist[ self::PROCESSOR_CRON_EVENT_NAME ] = self::MAX_PROCESSOR_JOB_COUNT;
```

Comment:
For some additional context, cron-control limits you to 10 total concurrent events as a default on each site. We should probably not take up half the queue w/ the default of 5 since this indexing can go on for quite a while. Perhaps something like:

```
$allowed_total_concurrency = ceil( Automattic\WP\Cron_Control\JOB_CONCURRENCY_LIMIT / 4 );
$whitelist[ self::PROCESSOR_CRON_EVENT_NAME ] = min( self::MAX_PROCESSOR_JOB_COUNT, $allowed_total_concurrency );
```

So the `MAX_PROCESSOR_JOB_COUNT` will never process much more than 25% of the overall concurrency limit. On larger sites that increase the total concurrency limit, eventually we can safely do all 5 at once. But by default, will do 3.